### DISCUSSION TITLE
How to trigger the OnNavigatedFrom method of a page manually？

### DISCUSSION THREAD

**Discussion by SongOfYouth** [microsoft/microsoft-ui-xaml#10217] [Q&A] [answered]
I want to notify a page to release some resource when close the TabViewItem which contains a frame which navigated to the page,  the OnNavigatingFrom method is only triggered when  the frame navigate to another page, is there any other way to trigger it?

**Comment by lhak** [comment, marked as answer]
The OnNavigatedFrom method of a page can be invoked by calling GetNavigationState on the frame object.

**Reply by SongOfYouth** [reply]
Thanks a lot. yes, it works! https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.frame.getnavigationstate?view=windows-app-sdk-1.6&devlangs=csharp&f1url=%3FappId%3DDev17IDEF1%26l%3DZH-CN%26k%3Dk(Microsoft.UI.Xaml.Controls.Frame.GetNavigationState)%3Bk(DevLang-csharp)%26rd%3Dtrue
